guides/micronaut-server-filter-request/java/src/test/java/example/micronaut/MemoryAppender.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example.micronaut;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.AppenderBase;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MemoryAppender extends AppenderBase<ILoggingEvent> {
+    private final List<ILoggingEvent> events = new ArrayList<>();
+
+    @Override
+    protected void append(ILoggingEvent e) {
+        synchronized (events) {
+            events.add(e);
+        }
```

Comment:
Instead of this, I believe we could do:

```java
    private final List<ILoggingEvent> events = Collections.synchronizedList(new ArrayList<>());
```

Which may be better advice for people following this guide?

Reply:
remove sync completely. Added groovy.